### PR TITLE
Add claims about all recipients and try to verify

### PR DESCRIPTION
**Pull request by azul**
Use new hookspec... now we can access recipients in
process_before_encryption.

Make sure to run with a current version of muacrypt.

We add claims for all recipients including their head imprints if we got them.
When receiving a message we then try and verify the own head imprint.

This is not the final logic. The own chain may already have progressed.
We actually care about key fingerprints more than headimprints and so
on.
Its just meant as one step towards finding claims in other peoples chains.

### CODE REVIEW

File: muacryptcc/plugin.py [hunk 1]
```diff
@@ -6,41 +6,91 @@
...
+        for recipient in recipients:
+            key = recipient
+            value = self.addr2root_hash[recipient]
+            peers_pk = self.addr2pk[recipient]
+            if peers_pk:
+                self.add_claim(claim=(key, value), access_pk=peers_pk)
+            else:
+                import logging ; logging.debug(recipient + " not found")
+        self.commit_to_chain()
+        msg["GossipClaims"] = pet2ascii(self.head)
```
Comment (bogdan-kulynych) [inline muacryptcc/plugin.py]: ``head`` should not be a petlib crypto object, but probably either raw bytes or hexdigest. I think this is not the right conversion to do.

Comment (azul) [reply muacryptcc/plugin.py]: ah... now i see where pet is coming from. It's raw bytes. So i guess some conversion to hex is what we want.

File: muacryptcc/plugin.py [hunk 2]
```diff
@@ -6,41 +6,97 @@
...
         self.store = store
         self.init_crypto_identity()
 
+    #
+    # muacrypt plugin hook implementations
+    #
+    @hookimpl
+    def process_incoming_gossip(self, addr2pagh, account_key, dec_msg):
+        root_hash = dec_msg["GossipClaims"]
+        store = FileStore(dec_msg["ChainStore"])
```
Comment (hpk42) [inline muacryptcc/plugin.py:42]: if either of GossipClaims or ChainStore is not set we should just return.

File: muacryptcc/plugin.py [hunk 3]
```diff
@@ -6,41 +6,97 @@
...
+            value = self.read_claim_from(peers_chain, recipient)
+            if value:
+                # for now we can only read claims about ourselves...
+                # so if we get a value it must be our head.
+                # TODO: pet2ascii is not the right thing here.
+                # What is it? Make sure to also use below.
+                assert value == pet2ascii(self.head)
+
+    @hookimpl
+    def process_before_encryption(self, sender_addr,
```
Comment (hpk42) [inline muacryptcc/plugin.py]: sender_addr should be in the next line for consistency.

File: muacryptcc/plugin.py [hunk 4]
```diff
@@ -6,41 +6,97 @@
...
+            sender_keyhandle,
+            recipient2keydata,
+            payload_msg,
+            _account):
+        recipients = recipient2keydata.keys()
+        if not recipients:
+            import logging ; logging.debug("no recipients found.\n")
```
Comment (hpk42) [inline muacryptcc/plugin.py]: please add logging at the top of the file, and just use logging.debug here.

**Pull request merged**